File: app/src/main/java/ai/elimu/content_provider/room/dao/VideoDao.java
package ai.elimu.content_provider.room.dao;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ai.elimu.content_provider.room.entity.Video;

@Dao
public interface VideoDao {

    @Insert
    void insert(Video video);

    @Query("SELECT * FROM Video WHERE id = :id")
    Video load(Long id);

    @Query("SELECT * FROM Video WHERE id = :id")
    Cursor loadAsCursor(Long id);

    @Query("SELECT * FROM Video")
    List<Video> loadAll();

    @Query("SELECT * FROM Video")
    Cursor loadAllAsCursor();

//    @Query("SELECT * FROM Video i WHERE i.id IN (SELECT Video_id FROM Video_Word WHERE words_id = :wordId)")
//    Cursor loadAllByWordLabelAsCursor(Long wordId);

    @Update
    void update(Video video);

    @Query("DELETE FROM Video")
    void deleteAll();
}

File: app/src/main/java/ai/elimu/content_provider/room/dao/VideoDao.kt
package ai.elimu.content_provider.room.dao

import ai.elimu.content_provider.room.entity.Video
import android.database.Cursor
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface VideoDao {
    @Insert
    fun insert(video: Video)

    @Query("SELECT * FROM Video WHERE id = :id")
    fun load(id: Long?): Video?

    @Query("SELECT * FROM Video WHERE id = :id")
    fun loadAsCursor(id: Long?): Cursor

    @Query("SELECT * FROM Video")
    fun loadAll(): MutableList<Video>

    @Query("SELECT * FROM Video")
    fun loadAllAsCursor(): Cursor

    @Query("SELECT * FROM Video WHERE title = :title")
    fun loadByTitleAsCursor(title: String?): Cursor

    @Update
    fun update(video: Video)

    @Query("DELETE FROM Video")
    fun deleteAll()
}
